#
#  Based on GCC 4.9 patches, set corresponding standards based
#  feature macros.  This may need to be revisited and coordinated
#  with additional header changes.  This is not being submitted
#  upstream until we have a clear path forward.
#
--- gcc-5.3.0/gcc/config/sol2.h.orig	Sat Jan 16 22:00:10 2016
+++ gcc-5.3.0/gcc/config/sol2.h	Sat Jan 16 22:02:58 2016
@@ -94,12 +94,26 @@
 	   library.  */					\
 	if (c_dialect_cxx ())				\
 	  {						\
-	    builtin_define ("__STDC_VERSION__=199901L");\
-	    builtin_define ("_XOPEN_SOURCE=600");	\
-	    builtin_define ("_LARGEFILE_SOURCE=1");	\
-	    builtin_define ("_LARGEFILE64_SOURCE=1");	\
-	    builtin_define ("__EXTENSIONS__");		\
-	  }						\
+            switch (cxx_dialect)			\
+              {			\
+              case cxx03: /* same as cxx98 */		\
+                builtin_define ("__STDC_VERSION__=199409L");\
+                builtin_define ("_XOPEN_SOURCE=500");	\
+                break;					\
+              case cxx11:				\
+              case cxx14:				\
+                builtin_define ("__STDC_VERSION__=199901L");\
+                builtin_define ("_XOPEN_SOURCE=700");	\
+                break;					\
+              default:					\
+	        builtin_define ("__STDC_VERSION__=199901L");\
+                builtin_define ("_XOPEN_SOURCE=600");	\
+                break;					\
+              }						\
+            builtin_define ("_LARGEFILE_SOURCE=1");	\
+            builtin_define ("_LARGEFILE64_SOURCE=1");	\
+            builtin_define ("__EXTENSIONS__=1");	\
+          }					        \
 	TARGET_SUB_OS_CPP_BUILTINS();			\
     } while (0)
 
 
